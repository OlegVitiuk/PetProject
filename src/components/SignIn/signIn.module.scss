.signInForm {
  width: 350px;
  margin-top: 5%;
  text-align: center;
}

.signInButton {
  width: 100%;
}

.signInWrapper {
  display: flex;
  justify-content: center;
  align-items: center;
}
