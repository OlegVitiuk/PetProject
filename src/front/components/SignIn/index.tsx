import { Button, Checkbox, Form, Icon, Input } from "antd";
import { FormComponentProps } from "antd/lib/form";
import { IAppState } from "front/store";
import * as React from "react";
import { connect } from "react-redux";
import styles from "./signIn.module.scss";

interface UserFormProps extends FormComponentProps {}

class SignIn extends React.Component<UserFormProps, any> {
  public handleSubmit = (e: React.SyntheticEvent) => {
    e.preventDefault();
    this.props.form.validateFields((err, values) => {
      if (!err) {
        console.log("Received values of form: ", values);
      }
    });
  };

  public render() {
    const { getFieldDecorator } = this.props.form;
    return (
      <div className={styles.signInWrapper}>
        <Form onSubmit={this.handleSubmit} className={styles.signInForm}>
          <h1>Sign In</h1>
          <Form.Item>
            {getFieldDecorator("username", {
              rules: [
                { required: true, message: "Please input your username!" }
              ]
            })(
              <Input
                prefix={
                  <Icon type="user" style={{ color: "rgba(0,0,0,.25)" }} />
                }
                placeholder="Username"
              />
            )}
          </Form.Item>
          <Form.Item>
            {getFieldDecorator("password", {
              rules: [
                { required: true, message: "Please input your Password!" }
              ]
            })(
              <Input
                prefix={
                  <Icon type="lock" style={{ color: "rgba(0,0,0,.25)" }} />
                }
                type="password"
                placeholder="Password"
              />
            )}
          </Form.Item>
          <Form.Item>
            <Button
              type="primary"
              htmlType="submit"
              className={styles.signInButton}
            >
              Log in
            </Button>
            Or <a href="">register now!</a>
          </Form.Item>
        </Form>
      </div>
    );
  }
}

// Grab the characters from the store and make them available on props
const mapStateToProps = (store: IAppState) => {
  return {};
};

export default connect(mapStateToProps)(
  Form.create<UserFormProps>({ name: "login" })(SignIn)
);
