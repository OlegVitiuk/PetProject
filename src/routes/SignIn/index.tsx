import { Button, Form, Icon, Input } from "antd";
import { FormComponentProps } from "antd/lib/form";
import * as React from "react";
import { connect } from "react-redux";
import { AppState } from "store";
import styles from "./signIn.module.scss";
import { Link } from "react-router-dom";
import { signIn } from "actions/authActions";
import { bindActionCreators, Action } from "redux";
import { ThunkDispatch } from "redux-thunk";
import { credsType } from "types/auth";

interface UserFormProps extends FormComponentProps {}

interface LinkDispatchProps {
  signIn: (creds: credsType) => void;
}

type Props = UserFormProps & LinkDispatchProps;

class SignIn extends React.Component<Props, any> {
  public handleSubmit = (e: React.SyntheticEvent) => {
    const { signIn } = this.props;

    e.preventDefault();
    this.props.form.validateFields((err, values) => {
      if (!err) {
        console.log("Received values of form: ", values);
        signIn(values);
      }
    });
  };

  public render() {
    const { getFieldDecorator } = this.props.form;
    return (
      <div className={styles.signInWrapper}>
        <Form onSubmit={this.handleSubmit} className={styles.signInForm}>
          <h1>Sign In</h1>
          <Form.Item>
            {getFieldDecorator("email", {
              rules: [
                { required: true, message: "Please input your email!" },
                {
                  type: "email",
                  message: "The input is not valid email!"
                }
              ]
            })(
              <Input
                prefix={
                  <Icon type="user" style={{ color: "rgba(0,0,0,.25)" }} />
                }
                placeholder="Email"
              />
            )}
          </Form.Item>
          <Form.Item>
            {getFieldDecorator("password", {
              rules: [
                { required: true, message: "Please input your Password!" }
              ]
            })(
              <Input
                prefix={
                  <Icon type="lock" style={{ color: "rgba(0,0,0,.25)" }} />
                }
                type="password"
                placeholder="Password"
              />
            )}
          </Form.Item>
          <Form.Item>
            <Button
              type="primary"
              htmlType="submit"
              className={styles.signInButton}
            >
              Log in
            </Button>
            Or <Link to="/sign-up">register now!</Link>
          </Form.Item>
        </Form>
      </div>
    );
  }
}

export default connect(
  null,
  (dispatch: ThunkDispatch<any, any, Action>): LinkDispatchProps => ({
    signIn: bindActionCreators(signIn, dispatch)
  })
)(Form.create<UserFormProps>({ name: "sign-in" })(SignIn));
