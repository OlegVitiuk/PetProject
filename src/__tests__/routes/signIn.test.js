import React from 'react';
import { shallow } from 'enzyme';
import SignIn from 'routes/SignIn';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const mockStore = configureMockStore([thunk]);

describe('Sign route', () => {
  let store = {};
  let wrapper;

  beforeEach(() => {
    store = mockStore({
      auth: {
        authError: 'asdfasfd'
      },
      firebase: {
        auth: {
          uuid: 'dff'
        }
      }
    });
  });

  it('render SignIn component', () => {
    const wrapper = shallow(<SignIn store={store} />);
    expect(wrapper).toMatchSnapshot();
  });
});
