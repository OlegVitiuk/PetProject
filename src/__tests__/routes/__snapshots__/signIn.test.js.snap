// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home route render Home component 1`] = `ShallowWrapper {}`;

exports[`Sign route render SignIn component 1`] = `ShallowWrapper {}`;
